api.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"github.com/hashicorp/vault/api"
+)
+
+// Store arbitrary string-data in Vault and return single-use wrapping token
+// Use the Vault response-wrapping features
+func Store(c *api.Client, ttl string, secrets map[string]interface{}) (string, error) {
+	// Set wrapped secrets ttl
+	c.SetWrappingLookupFunc(wrapItUp(ttl))
+
+	wrapped, err := c.Logical().Write("sys/wrapping/wrap", secrets)
+	if err != nil {
+		return "", err
+	}
+
+	return wrapped.WrapInfo.Token, err
+}
+
+// Retrieve data stored in token's wrapping path for the supplied Vault client/token
+func Retrieve(c *api.Client, token string) (map[string]interface{}, error) {
+
+	secrets, err := c.Logical().Unwrap(token)
+	if err != nil {
+		return nil, err
+	}
+	return secrets.Data, nil
+}
+
+func wrapItUp(ttl string) func(string, string) string {
```

Comment:
I would call this something like `staticWrappingLookup` or something along those lines.